### PR TITLE
feat: integrate new actor code with new vm

### PR DESCRIPTION
**Pull request by icorderi**
### Motivation

New things everywhere, we need to get them to talk to each other.

### Proposed changes

Delete all legacy actor code, bring in the new one.

### CODE REVIEW

**Comment by icorderi** [comment]
waiting on https://github.com/filecoin-project/specs-actors/pull/180

this can go in after that pending PR if the BLS check is removed.
Otherwise we need to fix BLS before landing first.

File: internal/pkg/vm/internal/vmcontext/vmcontext.go [hunk 1]
```diff
@@ -621,17 +621,23 @@ func msgCID(msg *types.UnsignedMessage) cid.Cid {
...
-		from:   vmaddr.SystemAddress,
-		to:     blockMiner,
-		value:  big.Zero(),
-		method: storagemining.ProcessVerifiedElectionPoStMethodID,
+		miner: blockMiner,
+		from:  vmaddr.SystemAddress,
+		to:    blockMiner,
+		value: big.Zero(),
+		// Dragons: figure out the new spec-actors place for this
+		// method: storagemining.ProcessVerifiedElectionPoStMethodID,
```
Comment (anorth) [inline internal/pkg/vm/internal/vmcontext/vmcontext.go]: This is no longer necessary. You can remove this method and the sending of the epost message.

File: internal/pkg/vm/testing.go [hunk 2]
```diff
@@ -222,25 +200,21 @@ type testStorage struct {
...
 	if cm, ok := v.(cbg.CBORMarshaler); ok {
 		buf := new(bytes.Buffer)
 		err := cm.MarshalCBOR(buf)
 		if err == nil {
-			nd, err := cbor.Decode(buf.Bytes(), types.DefaultHashFunction, -1)
-			if err != nil {
-				panic("failed to decode")
-			}
-			return nd.Cid()
+			return cid.NewCidV1(cid.Raw, buf.Bytes())
```
Comment (anorth) [inline internal/pkg/vm/testing.go]: I realise this is a test storage, but using raw-type CIDs here makes me nervous. I suggest using the cid.V1Builder with DAG_CBOR, as for other CIDs throughout gfc.

File: tools/gengen/util/gengen.go [hunk 3]
```diff
@@ -271,42 +261,43 @@ func setupMiners(vm consensus.GenesisVM, st state.Tree, keys []*types.KeyInfo, m
...
-
-			_, err := vm.ApplyGenesisMessage(addr, vmaddr.StoragePowerAddress, power.ProcessPowerReport, abi.NewTokenAmount(0), power.ProcessPowerReportParams{
-				Report:     powerReport,
-				UpdateAddr: mIDAddr,
-			})
-			if err != nil {
-				return nil, err
-			}
-		}
+		// Dragons: we need a new way of doing this
```
Comment (anorth) [inline tools/gengen/util/gengen.go]: fyi @ZenGround0 I think this is where you come in (but probably by messages for pre/committing sectors)

File: internal/pkg/types/keyinfo_test.go [hunk 4]
```diff
@@ -31,3 +31,16 @@ func TestKeyInfoMarshal(t *testing.T) {
 	assert.Equal(t, ki.Type(), kiBack.Type())
 	assert.True(t, ki.Equals(kiBack))
 }
+
+func TestBLSPublicKey(t *testing.T) {
+	tf.UnitTest(t)
+	t.Skip("Dragons: BLS is broken")
+
+	testKey, err := crypto.GenerateKey()
```
Comment (anorth) [inline internal/pkg/types/keyinfo_test.go]: This is not how you generate a BLS key. Use `bls.PrivateKeyGenerate()` and crypto_test.go.

File: tools/gengen/util/gengen.go (hunk 3 above)
Comment (ZenGround0) [reply tools/gengen/util/gengen.go]: 100% same page

File: internal/pkg/vm/internal/vmcontext/vmcontext.go [hunk 5]
```diff
@@ -441,11 +454,11 @@ func (vm *VM) applyMessage(msg *types.UnsignedMessage, onChainMsgSize uint32, mi
...
-		return message.Failure(exitcode.SysErrOutOfGas, gasTank.GasConsumed()), big.Zero()
+		return message.Failure(exitcode.SysErrOutOfGas, gasTank.GasConsumed()), big.Zero(), gasTank.GasConsumed().ToTokens(msgGasPrice)
 	}
 
 	// 2. Success!
-	return receipt, big.Zero()
+	return receipt, big.Zero(), gasTank.GasConsumed().ToTokens(msgGasPrice)
 }
 
 func (vm *VM) getMinerOwner(minerAddr address.Address) address.Address {
```
[the remark this answers is not in the document]
Comment (icorderi) [reply internal/pkg/vm/internal/vmcontext/vmcontext.go]: we can look at it later.

File: internal/pkg/vm/testing.go (hunk 2 above)
Comment (icorderi) [reply internal/pkg/vm/testing.go]: sure, that is just keeping whatever needed the old context working, this is all likely to get redone once we start adding the new vm tests.

**Pull request merged**